src/channel.js

```diff
@@ -141,14 +146,12 @@ export default class Channel {
    */
   _invokeListeners(options = {args: [], async: false}) {
     if (!this._mute) {
-      // ToDo: block adding/removing listeners to channel (throw an error) during dispatch operation
-      const listnersToInvoke = this._listeners.slice();
-      listnersToInvoke.forEach(listener => {
-        this._invokeListener(listener, options);
-        if (listener.once) {
-          this.removeListener(listener.callback, listener.context)
-        }
-      });
+      this._isDispatching = true;
+      const listenersToInvoke = this._listeners.slice();
+      listenersToInvoke.forEach(listener => this._invokeListener(listener, options));
```

Comment:
If some of sync listeners will throw an error, `isDispatching` will stay `true`.